deploy.sh

```diff
@@ -84,62 +147,99 @@ AKS_NAME=`echo ${BINDERHUB_NAME} | tr -cd '[:alnum:]-' | cut -c 1-59`-AKS
 
 # Get kubectl credentials from Azure
-echo "Fetching kubectl credentials from Azure"
-az aks get-credentials -n $AKS_NAME -g $RESOURCE_GROUP_NAME -o table
+echo "--> Fetching kubectl credentials from Azure"
+az aks get-credentials -n $AKS_NAME -g $RESOURCE_GROUP_NAME -o table | tee get-credentials.log
 
 # Check nodes are ready
-while [[ ! x`kubectl get node | awk '{print $2}' | grep Ready | wc -l` == x${AKS_NODE_COUNT} ]] ; do echo -n $(date) ; echo " : Waiting for all cluster nodes to be ready" ; sleep 15 ; done
+nodecount="$(kubectl get node | awk '{print $2}' | grep Ready | wc -l)"
+while [[ ${nodecount} -ne ${AKS_NODE_COUNT} ]] ; do echo -n $(date) ; echo " : ${nodecount} of ${AKS_NODE_COUNT} nodes ready" ; sleep 15 ; nodecount="$(kubectl get node | awk '{print $2}' | grep Ready | wc -l)" ; done
+echo
+echo "--> Cluster node status:"
+kubectl get node | tee kubectl-status.log
+echo
 
 # Setup ServiceAccount for tiller
-echo "Setting up tiller service account"
-kubectl --namespace kube-system create serviceaccount tiller
+echo "--> Setting up tiller service account"
+kubectl --namespace kube-system create serviceaccount tiller | tee tiller-service-account.log
 
 # Give the ServiceAccount full permissions to manage the cluster
-echo "Giving the ServiceAccount full permissions to manage the cluster"
-kubectl create clusterrolebinding tiller --clusterrole cluster-admin --serviceaccount=kube-system:tiller
+echo "--> Giving the ServiceAccount full permissions to manage the cluster"
+kubectl create clusterrolebinding tiller --clusterrole cluster-admin --serviceaccount=kube-system:tiller | tee cluster-role-bindings.log
 
 # Initialise helm and tiller
-echo "Initialising helm and tiller"
-helm init --service-account tiller --wait
+echo "--> Initialising helm and tiller"
+helm init --service-account tiller --wait | tee helm-init.log
 
 # Secure tiller against attacks from within the cluster
-echo "Securing tiller against attacks from within the cluster"
-kubectl patch deployment tiller-deploy --namespace=kube-system --type=json --patch='[{"op": "add", "path": "/spec/template/spec/containers/0/command", "value": ["/tiller", "--listen=localhost:44134"]}]'
+echo "--> Securing tiller against attacks from within the cluster"
+kubectl patch deployment tiller-deploy --namespace=kube-system --type=json --patch='[{"op": "add", "path": "/spec/template/spec/containers/0/command", "value": ["/tiller", "--listen=localhost:44134"]}]' | tee tiller-securing.log
+
+# Waiting until tiller pod is ready
+tillerStatus="$(kubectl get pods --namespace kube-system | grep ^tiller | awk '{print $3}')"
+while [[ ! x${tillerStatus} == xRunning ]] ; do echo -n $(date) ; echo " : tiller pod status : ${tillerStatus} " ; sleep 5 ; tillerStatus="$(kubectl get pods --namespace kube-system | grep ^tiller | awk '{print $3}')" ; done
```

Comment:
```suggestion
while [[ ! x${tillerStatus} == xRunning ]] ; do echo -n $(date) ; echo " : tiller pod status : ${tillerStatus} " ; sleep 30 ; tillerStatus="$(kubectl get pods --namespace kube-system | grep ^tiller | awk '{print $3}')" ; done
```
Wait 30 seconds for a ready tiller pod, instead of 5.